include/ignition/math/Region3.hh

```diff
@@ -0,0 +1,200 @@
+      /// it is defined with (i.e. Ix, Iy, Iz) are.
+      /// \return true if it is empty, false otherwise
+      public: bool Empty() const
+      {
+        return this->ix.Empty() || this->iy.Empty() || this->iz.Empty();
+      }
+
+      /// \brief Check if the region contains `_point`
+      /// \param[in] _point point to check for membership
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const Vector3<T> &_point) const
+      {
+        return (this->ix.Contains(_point.X()) &&
+                this->iy.Contains(_point.Y()) &&
+                this->iz.Contains(_point.Z()));
+      }
+
+      /// \brief Check if the region contains `_other` region
+      /// \param[in] _other region to check for membership
+      /// \return true if it is contained, false otherwise
+      public: bool Contains(const Region3<T> &_other) const
+      {
+        return (this->ix.Contains(_other.ix) &&
+                this->iy.Contains(_other.iy) &&
+                this->iz.Contains(_other.iz));
+      }
+
+      /// \brief Check if the region intersects `_other` region
+      /// \param[in] _other region to check for intersection
+      /// \return true if it is contained, false otherwise
+      public: bool Intersects(const Region3<T>& _other) const
+      {
+        return (this->ix.Intersects(_other.ix) &&
+                this->iy.Intersects(_other.iy) &&
+                this->iz.Intersects(_other.iz));
+      }
+
+      /// \brief Equality test operator
+      /// \param _other region to check for equality
+      /// \return true if regions are equal, false otherwise
+      public: bool operator==(const Region3<T> &_other) const
+      {
+        return this->Contains(_other) && _other.Contains(*this);
+      }
+
+      /// \brief Inequality test operator
+      /// \param _other region to check for inequality
+      /// \return true if regions are unequal, false otherwise
+      public: bool operator!=(const Region3<T> &_other) const
+      {
+        return !this->Contains(_other) || !_other.Contains(*this);
+      }
+
+      /// \brief Stream insertion operator
+      /// \param _out output stream
+      /// \param _r Region3 to output
+      /// \return the stream
+      public: friend std::ostream &operator<<(
+        std::ostream &_out, const ignition::math::Region3<T> &_r)
+      {
+        return _out <<_r.ix << " x " << _r.iy << " x " << _r.iz;
+      }
+
+      /// \brief The x-axis interval
+      private: Interval<T> ix;
+      /// \brief The y-axis interval
+      private: Interval<T> iy;
+      /// \brief The z-axis interval
+      private: Interval<T> iz;
+    };
+
+    namespace detail {
+       template<typename T>
+       const Region3<T> gUnboundedRegion3(
+           Interval<T>::Unbounded,
+           Interval<T>::Unbounded,
+           Interval<T>::Unbounded);
+    }
+    template<typename T>
+    const Region3<T> &Region3<T>::Unbounded = detail::gUnboundedRegion3<T>;
```

Comment:
Windows doesn't like this:

```
C:\Jenkins\workspace\ign_math-pr-win\ws\ign-math\include\ignition/math/Region3.hh(192): message : while compiling class template static data member 'const ignition::math::v7::Region3<double> &ignition::math::v7::Region3<double>::Unbounded' [C:\Jenkins\workspace\ign_math-pr-win\ws\build\ignition-math7\src\UNIT_Region3_TEST.vcxproj]

C:\Jenkins\workspace\ign_math-pr-win\ws\ign-math\src\Region3_TEST.cc(27): message : see reference to class template instantiation 'ignition::math::v7::Region3<double>' being compiled [C:\Jenkins\workspace\ign_math-pr-win\ws\build\ignition-math7\src\UNIT_Region3_TEST.vcxproj]

C:\Jenkins\workspace\ign_math-pr-win\ws\ign-math\include\ignition/math/Region3.hh(192,47): error C2491: 'ignition::math::v7::Region3<double>::Unbounded': definition of dllimport static data member not allowed [C:\Jenkins\workspace\ign_math-pr-win\ws\build\ignition-math7\src\UNIT_Region3_TEST.vcxproj]
```

Reply:
@chapulina https://github.com/ignitionrobotics/ign-math/pull/390/commits/b36e834a9b16521922f6ac6c0d936a9588fba8d1 should get rid of this.